Mathlib/Analysis/Calculus/MeanValue.lean

```diff
@@ -747,6 +747,126 @@ theorem exists_deriv_eq_slope : ∃ c ∈ Ioo a b, deriv f c = (f b - f a) / (b
   exists_hasDerivAt_eq_slope f (deriv f) hab hfc fun x hx =>
     ((hfd x hx).differentiableAt <| IsOpen.mem_nhds isOpen_Ioo hx).hasDerivAt
 
+/-- Lagrange's **Mean Value Theorem**, `deriv` version. -/
+theorem exists_deriv_eq_slope' : ∃ c ∈ Ioo a b, deriv f c = slope f a b := by
+  rw [slope_def_field]
+  exact exists_deriv_eq_slope f hab hfc hfd
+
+/-- A real function whose derivative tends to infinity from the right at a point is not
+differentiable on the right at that point -/
+theorem not_differentiableWithinAt_of_deriv_tendsto_atTop_Ioi (f : ℝ → ℝ) {a : ℝ}
+    (hf : Tendsto (deriv f) (𝓝[>] a) atTop) : ¬ DifferentiableWithinAt ℝ f (Ioi a) a := by
+  replace hf : Tendsto (derivWithin f (Ioi a)) (𝓝[>] a) atTop := by
```

Comment:
An alternative proof using Darboux theorem (e.g., [Set.OrdConnected.image_derivWithin](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Calculus/Darboux.html#Set.OrdConnected.image_derivWithin)): if `f'` tends to infinity, then `f` is differentiable on `[a, b]` for some `b > a` but `f' x > c > f' a` for all `a < x < b`, which is impossible because of Darboux thm. Also, this proof works the same way for left and right nhds.

Reply:
I had a quick look, but it doesn't seem significantly less painful than this proof, at least not enough to be worth a complete rewrite. Most of the pain in this version involves proving continuity on `Icc a b`, and that's easier than proving differentiability on the same interval.